src/blockMonitor/reorg.ts

```diff
@@ -201,8 +201,12 @@ export class ReorgDetector extends StartStopService {
     public async findCommonAncestor(
         newBlock: IBlockStub,
         currentHead: BlockStubChain
-    ): Promise<{ commonAncestor: BlockStubChain; differenceBlocks: IBlockStub[] }> {
-        let commonAncestor: BlockStubChain;
+    ): Promise<{ commonAncestor: BlockStubChain | null; differenceBlocks: IBlockStub[] }> {
+        if (newBlock.parentHash === null) {
```

Comment:
ArgumentError. I think you could even extend ArgumentError with a new error, ArgumentNullError and use that when checking for null.

Reply:
Changed to ArgumentError.